lib/twitter_cldr/utils/range_helpers.rb

```diff
@@ -0,0 +1,48 @@
+# encoding: UTF-8
+
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+require 'set'
+
+module TwitterCldr
+  module Utils
+
+    module RangeHelpers
+      def overlap?(range1, range2)
+        is_numeric_range?(range1) && is_numeric_range?(range2) && (
+          front_overlap?(range1, range2) ||
```

Comment:
Originally no, these overlap cases were all part of a single `#overlap?` method. During the development of this branch, I added a class called `RangeHash` and extracted the overlap cases out into the `RangeHelpers` module so `RangeHash` and `RangeSet` could share the functionality.  I ended up deleting `RangeHash` but left `RangeHelpers` by accident. I have moved the overlap methods back into `RangeSet` since there's no need to share them anymore. In the process, I discovered several places where it would be beneficial to use the distinct overlap cases.